x/fee_grant/module.go

```diff
@@ -0,0 +1,149 @@
+package fee_grant
+
+import (
+	"encoding/json"
+
+	"github.com/gorilla/mux"
+	"github.com/spf13/cobra"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	tx "github.com/cosmos/cosmos-sdk/x/fee_grant/internal/types/tx"
+)
+
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
+
+// AppModuleBasic defines the basic application module used by the fee_grant module.
+type AppModuleBasic struct{}
+
+// Name returns the fee_grant module's name.
+func (AppModuleBasic) Name() string {
+	return ModuleName
+}
+
+// RegisterCodec registers the fee_grant module's types for the given codec.
+func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
+	RegisterCodec(cdc)
+	tx.RegisterCodec(cdc)
+}
+
+// DefaultGenesis returns default genesis state as raw bytes for the fee_grant
+// module.
+func (AppModuleBasic) DefaultGenesis() json.RawMessage {
+	return []byte("[]")
```

Comment:
```suggestion
	return []byte("{}")
```

Reply:
```go
type GenesisState []FeeAllowanceGrant

func (a AppModuleBasic) getValidatedGenesis(bz json.RawMessage) (GenesisState, error) {
	var data GenesisState
	err := ModuleCdc.UnmarshalJSON(bz, &data)
	if err != nil {
		return nil, err
	}
	return data, data.ValidateBasic()
}
```

We unmarshal into an array, we must serialize an empty array - your suggestion would cause a parse error.